plugins/generator-1.15.2/forge-1.15.2/procedures/loot_table_items_foreach.java.ftl

```diff
@@ -0,0 +1,14 @@
+try {
+	if (${input$entity} instanceof ServerPlayerEntity) {
+		ServerPlayerEntity _ent = (ServerPlayerEntity) ${input$entity};
+		for (ItemStack itemstackiterator : _ent.server.getLootTableManager().getLootTableFromLocation(new ResourceLocation(${input$lootTable})).generate((new LootContext.Builder(_ent.getServerWorld())).withParameter(LootParameters.THIS_ENTITY, _ent)
+			.withNullableParameter(LootParameters.LAST_DAMAGE_PLAYER, (PlayerEntity) (_ent.getAttackingEntity() instanceof PlayerEntity ? _ent.getAttackingEntity() : null)).withNullableParameter(LootParameters.DAMAGE_SOURCE, _ent.getLastDamageSource())
+			.withNullableParameter(LootParameters.KILLER_ENTITY, _ent.getLastDamageSource().getTrueSource()).withNullableParameter(LootParameters.DIRECT_KILLER_ENTITY, _ent.getLastDamageSource().getImmediateSource())
+			.withParameter(LootParameters.POSITION, new BlockPos(_ent)).withParameter(LootParameters.BLOCK_STATE, _ent.getServerWorld().getBlockState(new BlockPos(_ent))).withNullableParameter(LootParameters.BLOCK_ENTITY, _ent.getServerWorld().getTileEntity(new BlockPos(_ent)))
+			.withParameter(LootParameters.TOOL, _ent.inventory.getCurrentItem()).withParameter(LootParameters.EXPLOSION_RADIUS, 0F).withRandom(_ent.getRNG()).withLuck(_ent.getLuck()).build(LootParameterSets.GENERIC))) {
+			${statement$foreach}
+		}
+	}
+} catch (Exception _e) {

```

Comment:
Why? Do proper checks so this does not happen, not just blindly catch-all exceptions.

Reply:
It's made to avoid ResourceLocationException, I just forgot about `tryCreate()` for some reason. Thanks for your catch.

Reply:
It's made to avoid ResourceLocationException, I just forgot about `tryCreate()` for some reason. Thanks for your catch.